Docs/Add standard json schema examples

Eomm opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Tip: `npm run bench` to compare branches interactively.

Contributors guide: https://github.com/fastify/fastify/blob/master/CONTRIBUTING.md
-->

#### Checklist

- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

Hi,
I'm opening this PR to get some feedback about this changing for issue #1329 .

I have written a "big" example with fluent schema because I think that put many little examples could be too broad and more confusing.

I have split the first example to view better the four main parameters of schema validation input (`body, querystring, params and headers`).

There are some questions I would like to expose you:
- `fluent-schema` has started a [big refactor](https://github.com/fastify/fluent-schema/issues/22) for the v1 release, so I will update docs when it will be released
- the `definitions` and `$ref` tags, as discussed in other issues like #1351 is supported by `fluent-schema` but not in fastify: should we implement that function or add a disclaimer in docs?

Let me know what do you think and what I should I do 😄 
Thank you

PS: next days I will not have the PC so I'll continue with this PR at starts of January 💪

mcollina commented on docs/Validation-and-Serialization.md:
```diff
@@ -250,9 +279,47 @@ fastify.setErrorHandler(function (error, request, reply) {
 - [JSON Schema](http://json-schema.org/)
 - [Understanding JSON schema](https://spacetelescope.github.io/understanding-json-schema/)
 - [fast-json-stringify documentation](https://github.com/fastify/fast-json-stringify)
+- [fluent-schema documentation](https://github.com/fastify/fluent-schema)
```
is this link used somewhere?

Eomm replied:
I used it here:
https://github.com/fastify/fastify/pull/1355/commits/5fdcf692da611b5413062650570e639bbfe45027#diff-71112d6db09dd080118d76d15c362f4eR284

But maybe is redundant?

Eomm commented:
> we could support those, but we should do in a separate PR (if you would like to do it, please send it over!)

Perfect, I'll do it!

allevo commented:
I'm not so convinced to introduce this kind of suggestion into the core documentation: maybe a simple link to the repository or npm package is enough. 

For example we introduce `Joi` with a very sort example [here](https://www.fastify.io/docs/latest/Validation-and-Serialization/#schema-compiler).

Eomm commented:
Hi, I'm back 😄 Happy new year 🎉 

Regarding this PR:
I saw that the refactor of [fluent-schema booming](https://github.com/fastify/fluent-schema/pull/24)

so, could be a good trade-off adds to docs more json-schema standard examples?
Then, with a future PR, could we add fluent schema when it will be released with the new API and fixes?

mcollina commented:
I'm good with that approach!

Eomm commented:
Hi stale bot 😋
I'm waiting for the review: as shared I reverted the fluent schema example (monitoring that repo for the new release) and added to this pr only the json schema standards

aboutlo commented:
Meanwhile, fluent-schema v0.6.0 is out https://github.com/fastify/fluent-schema/releases/tag/v0.6.0 :)

Eomm commented:
Updated the title to match this https://github.com/fastify/fastify/pull/1355#issuecomment-451581620
